File: optimize_fuel_efficiency.c
#include <stdio.h>

double optimize_fuel_efficiency(double thrust, double drag, double sfc) {
    if (thrust <= 0.0 || drag <= 0.0 || sfc <= 0.0) {
        return -1.0;
    }
    return thrust / (drag * sfc);
}

File: fortran/optimize_fuel_efficiency.f90
module efficiency_module
  implicit none
contains
  function optimize_fuel_efficiency(thrust, drag, sfc) result(efficiency)
    ! Input: thrust (N), drag (N), sfc (kg/Nh)
    ! Output: fuel efficiency (km/kg)

    real(8), intent(in) :: thrust, drag, sfc
    real(8) :: efficiency

    if (drag <= 0.0d0 .or. sfc <= 0.0d0) then
      efficiency = -1.0d0
    else
      efficiency = thrust / (drag * sfc)
    end if
  end function optimize_fuel_efficiency
end module efficiency_module
